### PR TITLE
[FEATURE] 8LTS: Replace deprecated TSFE->anchorPrefix

### LINKED ISSUE #223: $TSFE->anchorPrefix is deprecated
> $TSFE->anchorPrefix deprecated since TYPO3 CMS 7, will be removed with TYPO3 CMS 8, use substr(GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL'), strlen(GeneralUtility::getIndpEnv('TYPO3_SITE_URL'))) directly

> Alternatively, use config.absRefPrefix to achieve the same result.

while `\TYPO3\CMS\Core\Utility\GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL')` seems to be available in TYPO3 6.2, so this can be made in metaseo 2.x.

https://wiki.typo3.org/TYPO3.CMS/Releases/7.2/Deprecation#Deprecation:_.2365938_-_Discourage_usage_of_.22.24TSFE-.3EanchorPrefix.22

### PR DESCRIPTION
**Pull request by thomaszbz**
Fixes #223

### CODE REVIEW

**Comment by thomaszbz** [comment]
not tested yet.

File: Classes/Utility/FrontendUtility.php [hunk 1]
```diff
@@ -185,21 +185,21 @@ public static function isCacheable($conf = null)
...
+     * @return string
      */
     public static function getCurrentUrl()
     {
-        $ret = null;
-
-        $TSFE = self::getTsfe();
+        //former $TSFE->anchorPrefix got deprecated in TYPO3 7.x
+        $tsfeAnchorPrefix = substr(
+            Typo3GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL'),
```
Comment (simonschaufi) [inline Classes/Utility/FrontendUtility.php]: I would save it in a variable to make it a little bit more faster.

Comment (thomaszbz) [inline Classes/Utility/FrontendUtility.php]: Hm... delivering the current Url is the kind of task a CMS should provide. Caching included.

Comment (simonschaufi) [inline Classes/Utility/FrontendUtility.php]: I just mean some small optimization with:
```php
$requestUrl = Typo3GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL');
$tsfeAnchorPrefix = substr($requestUrl, strlen($requestUrl));
```

Comment (thomaszbz) [inline Classes/Utility/FrontendUtility.php]: Well... yes. Can be done.

Comment (thomaszbz) [inline Classes/Utility/FrontendUtility.php]: Revisiting this, I come to the conclusion that `'TYPO3_REQUEST_URL'` not equals `'TYPO3_SITE_URL'`.

Would be wrong to use $requestUrl twice here.

Comment (simonschaufi) [inline Classes/Utility/FrontendUtility.php]: oh, i totally overlooked this!

Comment (thomaszbz) [inline Classes/Utility/FrontendUtility.php]: But for `'TYPO3_SITE_SCRIPT'`, TYPO3 does a

    case 'TYPO3_SITE_SCRIPT':
        $retVal = substr(self::getIndpEnv('TYPO3_REQUEST_URL'), strlen(self::getIndpEnv('TYPO3_SITE_URL')));

... which is pretty much the same, just shorter.

**Comment by thomaszbz** [comment]
Still good to see someone takes care on changes!

Merged this now via  PR #481 

@simonschaufi Would be great if you could counter test feature/8lts for regressions. We're releasing 3.0.0 tomorrow (that's the plan at least).

**Pull request closed without merging**